src/glview/GLView.cc

```diff
@@ -219,6 +219,23 @@ void GLView::paintGL()
   }
   glDisable(GL_LIGHTING);
   if (showaxes) GLView::showSmallaxes(axescolor);
+
+  // Workaround for inconsistent QT behavior related to handling custom OpenGL widgets that
+  // leave non opaque alpha values in final output.
+  // On wayland that can cause window to become transparent or blurry trail effect in the
+  // parts that contain partially transparent objects.
+  //
+  // At the end of rendering clear alpha value, so that it doesn't matter how rest of the
+  // compositing stack at QT and desktop level would interpret transparent pixels.
+  //
+  // Solves https://github.com/openscad/openscad/issues/3689.
+  //
+  // Originally developed by @karliss for FreeCAD (https://github.com/FreeCAD/FreeCAD/pull/19499).
+  glPushAttrib(GL_COLOR_BUFFER_BIT);
+  glColorMask(false, false, false, true);
+  glClearColor(0, 0, 0, 1);
+  glClear(GL_COLOR_BUFFER_BIT);
+  glPopAttrib();
```

Comment:
I don't think you need to push/pop here. We're trying to move off immediate-mode OpenGL, so that we can move 
 to OpenGL > 3 or GLES, so it would be good to avoid introducing new code using it.

Reply:
without push/pop preview is completely black

Reply:
that seems to work!